src/QuizAtom.tsx

```diff
@@ -1,7 +1,268 @@
+
+    return (
+        <div
+            className={css`
+                ${body.medium()};
+            `}
+        >
+            <fieldset className={fieldsetStyle}>
+                <div className={questionStyle}>
+                    <legend>
+                        <span>{number + '.'}</span>
+                        {text}
+                    </legend>
+                </div>
+                {imageUrl && (
+                    <img
+                        className={css`
+                            width: 100%;
+                        `}
+                        src={imageUrl}
+                    ></img>
+                )}
+                <div>
+                    {answers.map((answer) => (
+                        <Answer
+                            key={answer.id}
+                            id={answer.id}
+                            text={answer.text}
+                            isCorrect={answer.isCorrect}
+                            revealText={answer.revealText}
+                            isChosen={chosen === answer.id}
+                            setChosen={setChosen}
+                            isAnswered={chosen !== undefined}
+                        />
+                    ))}
+                </div>
+            </fieldset>
+        </div>
+    );
+};
+
+type AnswerProps = {
+    isChosen: boolean;
+    setChosen: React.Dispatch<React.SetStateAction<string | undefined>>;
+    isAnswered: boolean;
+};
+
+export const Answer = ({
+    id,
+    text,
+    isCorrect,
+    revealText,
+    isChosen,
+    setChosen,
+    isAnswered,
+}: AnswerType & AnswerProps): JSX.Element => {
+    const handleClick = () => setChosen(id);
+    const handleKeyPress = (e: KeyboardEvent) => {
+        if (e.key === 'Enter' && !isAnswered) {
+            setChosen(id);
+        }
+    };
+    return (
+        <div
+            className={css`
+                clear: both;
+                margin-bottom: 6px;
+                position: relative;
+            `}
+        >
+            <input
+                type="radio"
+                tabIndex={-1}
+                id={`answer-${id}`}
+                required
+                defaultChecked={isChosen}
+                onClick={handleClick}
+                disabled={isAnswered}
+                className={css`
+                    ${visuallyHidden};
+                `}
+            ></input>
+            <label
+                htmlFor={`answer-${id}`}
+                tabIndex={0}
+                className={cx({
+                    [commonLabelStyle]: true,
+                    [focusLabelStyle]: !isAnswered,
+                    [opaqueGreenBackgroundColour]: isCorrect && isChosen,
+                    [translucentGreenBackgroundColour]:
+                        isCorrect && !isChosen && isAnswered,
+                    [redBackgroundColour]: !isCorrect && isChosen,
+                    [neutralBackgroundColour]: !isCorrect && !isChosen,
+                })}
```

Comment:
the css here is quite hard to follow. I think `Answer` as a component this might be a good candidate to split out differently.

This is just an idea, but it might be worth splitting out:

- `SelectedAwnser`
- `CorrectAwnser`
- `WrongAwnser`
- `PendingAwnser`

you could then have an emotion css class for each and not need to worry too much about colour combinations.

Also this means we could also avoid having nested ternary as each component would be stateless.